### PR TITLE
feat(torch): allow data augmentation w/o db

### PR DESCRIPTION
**Pull request by fantes**
plus cleanup around image_to_tensor

### CODE REVIEW

**Review by royale — APPROVED** [review]

File: src/backends/torch/torchdataset.cc [hunk 1]
```diff
@@ -366,13 +371,86 @@ namespace dd
     return d.at(i).sizes().vec();
   }
 
+  void TorchDataset::dataaug_then_push(const cv::Mat &bgr,
```
Comment (Bycob) [inline src/backends/torch/torchdataset.cc:374]: dataaug_then_push_back (at first I thought this method was pushing to db)

File: src/backends/torch/torchdataset.cc [hunk 2]
```diff
@@ -366,13 +371,86 @@ namespace dd
...
+              }
+            else if (_segmentation)
+              _img_rand_aug_cv.augment_test_with_segmap(bgr_sample,
+                                                        bw_target_sample);
+            else
+              _img_rand_aug_cv.augment_test(bgr_sample);
+          }
+
+        torch::Tensor imgt
+            = image_to_tensor(bgr_sample, bgr_sample.rows, bgr_sample.cols);
```
Comment (Bycob) [inline src/backends/torch/torchdataset.cc:420]: It looks like "rows" and "cols" parameters are useless, they can be deduced from the cv::Mat. So we can drop them from image_to_tensor?

Comment (fantes) [reply src/backends/torch/torchdataset.cc:420]: I dont not think it is always the case that we use first param dims when calling image_to_tensor, see for instance torchdataset.cc:260 and torchdataset.cc:279 (ie add_image_batch flavors that are called from inputconn)

Comment (fantes) [reply src/backends/torch/torchdataset.cc]: done

**Review by Bycob — APPROVED** [review]

**Pull request merged**